sdk/spring/spring-cloud-azure-actuator-autoconfigure/pom.xml

```diff
@@ -66,6 +66,12 @@
       <version>4.38.0</version> <!-- {x-version-update;com.azure:azure-cosmos;dependency} -->
       <optional>true</optional>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-spring-data-cosmos</artifactId>
+      <version>3.29.0</version> <!-- {x-version-update;com.azure:azure-spring-data-cosmos;dependency} -->
+      <optional>true</optional>
+    </dependency>
```

Comment:
And if you move the bean configuration to `CosmosDataAutoConfiguration`, do we still need this?